src/LanguageServer/Protocol/Features/Options/SolutionCrawlerOptionsStorage.cs

```diff
@@ -75,14 +57,7 @@ public static CompilerDiagnosticsScope GetBackgroundCompilerAnalysisScope(this I
             return CompilerDiagnosticsScope.VisibleFilesAndOpenFilesWithPreviouslyReportedDiagnostics;
         }
 
-        return globalOptions.GetOption(SolutionBackgroundAnalysisScopeOption) switch
-        {
-            BackgroundAnalysisScope.VisibleFilesAndOpenFilesWithPreviouslyReportedDiagnostics => CompilerDiagnosticsScope.VisibleFilesAndOpenFilesWithPreviouslyReportedDiagnostics,
-            BackgroundAnalysisScope.OpenFiles => CompilerDiagnosticsScope.OpenFiles,
-            BackgroundAnalysisScope.FullSolution => CompilerDiagnosticsScope.FullSolution,
-            BackgroundAnalysisScope.None => CompilerDiagnosticsScope.None,
-            _ => globalOptions.GetOption(CompilerDiagnosticsScopeOption, language),
-        };
+        return globalOptions.GetOption(CompilerDiagnosticsScopeOption, language);
```

Comment:
>return globalOptions.GetOption(CompilerDiagnosticsScopeOption, language); [](http://example.com/codeflow?start=8&length=73)

This isn't clear to me. Seems like this is changing behavior as it matches the default case which would never have hit before.

Reply:
SolutionBackgroundAnalysisScopeOption was normally *null*.  so it would fall through to that default case.